internal/api/routes.go

```diff
@@ -9,19 +9,30 @@ import (
 	"github.com/edulinq/autograder/internal/api/logs"
 	"github.com/edulinq/autograder/internal/api/static"
 	"github.com/edulinq/autograder/internal/api/users"
+	"github.com/edulinq/autograder/internal/util"
 )
 
-var baseRoutes = []*core.Route{
+var baseRoutes = []core.Route{
 	core.NewRedirect("GET", ``, `/static/index.html`),
 	core.NewRedirect("GET", `/`, `/static/index.html`),
 	core.NewRedirect("GET", `/index.html`, `/static/index.html`),
 
-	core.NewRoute("GET", `/static`, static.Handle),
-	core.NewRoute("GET", `/static/.*`, static.Handle),
+	core.NewBaseRoute("GET", `/static`, static.Handle),
+	core.NewBaseRoute("GET", `/static/.*`, static.Handle),
 }
 
-func GetRoutes() *[]*core.Route {
-	routes := make([]*core.Route, 0)
+type APIDescription struct {
+	Endpoints map[string]map[string]EndpointDescription
```

Comment:
Put some JSON info on these fields.

The idea is that we just convert the entire description to JSON.
This way, we can later just put this field in our API response type and we're done.